libs/SmartSync/src/com/salesforce/androidsdk/smartsync/manager/SyncManager.java

```diff
@@ -337,7 +335,44 @@ private void syncUp(SyncState sync, SyncUpdateCallback callback) throws Exceptio
         }
 	}
 
-    private boolean syncUpOneRecord(String soupName, List<String> fieldlist, JSONObject record) throws JSONException, IOException {
+    private boolean isNewerThanServer(String objectType, String objectId,
```

Comment:
Maybe there's a better way of doing this, rather than fetch it per record. The best I can think of is batching them per record type. If there's a better suggestion, I'm open to ideas.